### PR TITLE
Refactor graphqlcontext since getContext() is deprecated

### PR DESCRIPTION
**Pull request by sandrawangyx**
Signed-off-by: Sandra Wang <yx97.wang@gmail.com>

## PR description
This PR is focused on resolving the deprecated dependency of graphQL context.

Deprecated: `graphql.schema.DataFetchingEnvironment#getContext`

## Fixed Issue(s)
Issue #2776

### CODE REVIEW

**Comment by sandrawangyx** [comment]
I refactored graphqlcontext which I believe is one of the critical pieces in the code base.
all unit tests passed except for this one: 
``` 
org.hyperledger.besu.ethereum.api.jsonrpc.JsonRpcHttpServiceTlsMisconfigurationTest > exceptionRaisedWhenInvalidKeystoreFileIsSpecified FAILED
    java.lang.AssertionError at JsonRpcHttpServiceTlsMisconfigurationTest.java:193
```

the error message is:
```
java.lang.AssertionError: 
Expecting throwable message:
  "org.hyperledger.besu.ethereum.api.jsonrpc.JsonRpcServiceException: Failed to bind Ethereum JSON-RPC listener to 127.0.0.1:0: java.io.IOException: Tag number over 30 is not supported"
to contain:
  "Short read of DER length"
but did not.
```

**Comment by macfarla** [comment]
> I refactored graphqlcontext which I believe is one of the critical pieces in the code base. all unit tests passed except for this one:
> 
> ```
> org.hyperledger.besu.ethereum.api.jsonrpc.JsonRpcHttpServiceTlsMisconfigurationTest > exceptionRaisedWhenInvalidKeystoreFileIsSpecified FAILED
>     java.lang.AssertionError at JsonRpcHttpServiceTlsMisconfigurationTest.java:193
> ```
> 
> the error message is:
> 
> ```
> java.lang.AssertionError: 
> Expecting throwable message:
>   "org.hyperledger.besu.ethereum.api.jsonrpc.JsonRpcServiceException: Failed to bind Ethereum JSON-RPC listener to 127.0.0.1:0: java.io.IOException: Tag number over 30 is not supported"
> to contain:
>   "Short read of DER length"
> but did not.
> ```
> 
> Should I change the expected message content?

I get this same failure with the different error message when running the test locally. But it seems to be passing in CI so no don't change the message.

File: besu/src/main/java/org/hyperledger/besu/RunnerBuilder.java [hunk 1]
```diff
@@ -655,13 +655,12 @@ public Runner build() {
...
-              protocolSchedule,
-              transactionPool,
-              miningCoordinator,
-              synchronizer);
+      final Map<String, Object> graphQlContextMap = new ConcurrentHashMap<>();
+      graphQlContextMap.putIfAbsent("blockchainQueries", blockchainQueries);
+      graphQlContextMap.putIfAbsent("protocolSchedule", protocolSchedule);
+      graphQlContextMap.putIfAbsent("transactionPool", transactionPool);
+      graphQlContextMap.putIfAbsent("miningCoordinator", miningCoordinator);
+      graphQlContextMap.putIfAbsent("synchronizer", synchronizer);
```
Comment (macfarla) [inline besu/src/main/java/org/hyperledger/besu/RunnerBuilder.java:663]: looks fine overall. One change I would suggest is to make these strings constants eg BLOCKCHAIN_QUERIES. Maybe a new enum in the graphql package eg see GraphQLResponseType.

**Review by macfarla — COMMENTED** [review]
Looks good overall, one suggestion to add an enum to avoid repeating the string constants.

File: ethereum/api/src/main/java/org/hyperledger/besu/ethereum/api/graphql/GraphQLContextType.java [hunk 2]
```diff
@@ -0,0 +1,25 @@
+/*
+ * Copyright ConsenSys AG.
```
Comment (jflo) [inline ethereum/api/src/main/java/org/hyperledger/besu/ethereum/api/graphql/GraphQLContextType.java:2]: I'm sure this was just copied from elsewhere in the code, but we are trying to make sure new files are `Copyright Hyperledger Besu Contributors` moving forward.

**Review by jflo — COMMENTED** [review]
Minor, non-blocking note. Welcome to the project!

File: ethereum/api/src/main/java/org/hyperledger/besu/ethereum/api/graphql/GraphQLContextType.java (hunk 2 above)
Comment (sandrawangyx) [reply ethereum/api/src/main/java/org/hyperledger/besu/ethereum/api/graphql/GraphQLContextType.java:2]: thank you @jflo , will update!

**Review by macfarla — APPROVED** [review]
Nice! Thanks for your contribution!

**Comment by Neurone** [comment]
> > I refactored graphqlcontext which I believe is one of the critical pieces in the code base. all unit tests passed except for this one:
> > ```
> > org.hyperledger.besu.ethereum.api.jsonrpc.JsonRpcHttpServiceTlsMisconfigurationTest > exceptionRaisedWhenInvalidKeystoreFileIsSpecified FAILED
> >     java.lang.AssertionError at JsonRpcHttpServiceTlsMisconfigurationTest.java:193
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > the error message is:
> > ```
> > java.lang.AssertionError: 
> > Expecting throwable message:
> >   "org.hyperledger.besu.ethereum.api.jsonrpc.JsonRpcServiceException: Failed to bind Ethereum JSON-RPC listener to 127.0.0.1:0: java.io.IOException: Tag number over 30 is not supported"
> > to contain:
> >   "Short read of DER length"
> > but did not.
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > Should I change the expected message content?
> 
> I get this same failure with the different error message when running the test locally. But it seems to be passing in CI so no don't change the message.

I encountered the same error building Besu just now: "Tag number over 30 is not supported".

JDK 17 added that error more than one year ago; see here: https://github.com/openjdk/jdk/commit/3d2b4cc567777d0849818f5a4cc8fc6394a55f8d 

I suggest updating the test (and the JDK on the CI machine also ^^); it seems weird to me to have an always failing test.

**Pull request merged**